### PR TITLE
Update CMakeLists.txt to be used through pkg-config

### PR DESCRIPTION
**Pull request by mayataka**
Hi @jmirabel

Thank you for developing a useful repository.
I want to use `pinocchio-gepetto-viewer` (specifically class `pinocchio::gepetto::Viewer`) through pkg-config or CMake, and this pull request is a modification of CMakeLists.txt for pkg-config.
I also added include guards in .hpp and .hxx headers and macro for HPP-FCL in .cc file.

Best,

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,47 +1,79 @@
...
+  ${PROJECT_NAME} 
+  PUBLIC 
+  $<INSTALL_INTERFACE:include>
+)
+IF(PINOCCHIO_USE_HPP_FCL)
+  target_compile_definitions(
+    ${PROJECT_NAME} 
+    PRIVATE
+    PINOCCHIO_WITH_HPP_FCL
+  )
```
Comment (jmirabel) [inline CMakeLists.txt]: @jcarpent I would expect this flag to come with target `pinocchio::pinocchio`. Any reason this is not the case ?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,47 +1,79 @@
...
-if(gepetto-viewer_FOUND)
-  CHECK_MINIMAL_CXX_STANDARD(11 ENFORCE)
-  add_subdirectory(plugins)
-endif()
+IF(BUILD_PLUGINS)
+  IF(gepetto-viewer_FOUND)
+   CHECK_MINIMAL_CXX_STANDARD(11 ENFORCE)
+   ADD_SUBDIRECTORY(plugins)
+  ENDIF()
+ENDIF()
```
Comment (jmirabel) [inline CMakeLists.txt]:
```suggestion
IF(BUILD_PLUGINS)
  ADD_PROJECT_DEPENDENCY(gepetto-viewer REQUIRED)
  CHECK_MINIMAL_CXX_STANDARD(11 ENFORCE)
  ADD_SUBDIRECTORY(plugins)
ENDIF()
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,47 +1,79 @@
...
+PROJECT(${PROJECT_NAME} ${PROJECT_ARGS})
+
+OPTION(BUILD_TESTS "Build tests" ON)
+OPTION(BUILD_PLUGINS "Build plugins" OFF)
 
 # Add a required dependency
-add_project_dependency(gepetto-viewer)
-add_project_dependency(gepetto-viewer-corba REQUIRED)
-add_project_dependency(pinocchio REQUIRED)
+ADD_PROJECT_DEPENDENCY(gepetto-viewer)
```
Comment (jmirabel) [inline CMakeLists.txt]:
```suggestion
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,47 +1,79 @@
...
   ${${PROJECT_NAME}_HEADERS}
-  src/viewer.cc)
-target_link_libraries(pinocchio-gepetto-viewer PUBLIC pinocchio::pinocchio gepetto-viewer-corba::gepetto-viewer-corba)
+  ${PROJECT_SOURCE_DIR}/src/viewer.cc
+)
+TARGET_LINK_LIBRARIES(
+  ${PROJECT_NAME}
+  PUBLIC 
+  pinocchio::pinocchio 
+  gepetto-viewer::gepetto-viewer
```
Comment (jmirabel) [inline CMakeLists.txt]:
```suggestion
```

File: CMakeLists.txt (hunk 2 above)
Comment (jmirabel) [reply CMakeLists.txt]: With option `BUILD_PLUGIN`, gepetto-viewer becomes mandatory.

**Review by jmirabel — CHANGES_REQUESTED** [review]
In CMakeLists, many changes are purely cosmetic (mostly capitalization) which makes it hard to know what actually changed. Moreover, I tend to prefer the lower case version (although the original file was not homogeneous).

I would prefer that you revert the cosmetic changes, but I can understand if you don't have time for this.

File: CMakeLists.txt (hunk 1 above)
Comment (jcarpent) [reply CMakeLists.txt]: This is indeed exported. Please see the content of `pinocchioTargets.cmake`.

Comment (jcarpent) [reply CMakeLists.txt]: This is also exported in the `pinocchio.pc`

Comment (jmirabel) [reply CMakeLists.txt]: Ok, so @mayataka, you don't need to pass this flag. It is done automatically when needed.

Comment (mayataka) [reply CMakeLists.txt]: > In CMakeLists, many changes are purely cosmetic (mostly capitalization) which makes it hard to know what actually changed. Moreover, I tend to prefer the lower case version (although the original file was not homogeneous).
> 
> I would prefer that you revert the cosmetic changes, but I can understand if you don't have time for this.

Sorry for not noticing it.
Then, what I should do is to revert the cosmetic commit and re-commit only changes, right?

Comment (mayataka) [reply CMakeLists.txt]: > Ok, so @mayataka, you don't need to pass this flag. It is done automatically when needed.

I see. I remove its redundant block.

**Comment by jmirabel** [comment]
> Sorry for not noticing it.
> Then, what I should do is to revert the cosmetic commit and re-commit only changes, right?

Yes, if possible, that would be appreciated.

**Comment by mayataka** [comment]
> > Sorry for not noticing it.
> > Then, what I should do is to revert the cosmetic commit and re-commit only changes, right?
> 
> Yes, if possible, that would be appreciated.

I see. I'll also incorporate your change requests in the new commit.

**Comment by jmirabel** [comment]
Thank you for your contrib.

**Pull request merged**